sageopt/coniclifts/constraints/set_membership/pow_cone.py

```diff
@@ -0,0 +1,231 @@
+import numpy as np
+from sageopt.coniclifts.constraints.set_membership.setmem import SetMembership
+from sageopt.coniclifts.base import Expression, ScalarVariable, Variable
+from sageopt.coniclifts.cones import Cone
+from sageopt.coniclifts.operators.norms import vector2norm
+from sageopt.coniclifts.problems.problem import Problem
+
+
+class PowCone(SetMembership):
+    """
+    A power cone constraint is defined with an n-dimensional vector
+    :math:'\\alpha' which is elementwise positive and
+    sums to 1. The cone can be summarized by writing that
+    :math:'(w, z) \\in C_{\\mathrm{power}}' if:
+
+    .. math::
+        \\prod_{i=0}^n w_i^\\alpha_i > \\| z \\|
+
+    where :math:'w' is an n dimensional vector and :math:'z' is a scalar.
+
+    Parameters
+    ----------
+
+    w : Expression
+
+        An (n+1)-dimensional vector subject to this Power Cone constraints.
+        One element of this vector is z which is
+        determined by the elements of
+
+    lamb: ndarray
+        An (n+1)-dimensional array with 1 negative entry and n positive entries
+        whose sum is 0. The n positive entries determine the elements of
+        :math:'\\alpha' while the negative entry of lamb determines the index of
+        z within the w parameter.
+
+
+    Attributes
+    ----------
+    id: int
+        The id of the Power Cone constraint object
+
+    w : Expression
+        The inputted (n+1)-dimensional vector inputted
+        into the Power Cone constructor
+
+    lamb : ndarray
+        The inputted (n+1)-dimensional array inputted into the
+        Power Cone constructor that sums to 0
+
+    w_low : Expression
+        An n-dimensional expression used to describe w in the Power Cone
+
+    z_low : Expression
+        A scalar expression used to describe z in the Power Cone
+
+    alpha : ndarray
+        The n-dimensional vector :math:'\\alpha' that sums to 1
+        that defines the power Cone constraint
+
+    Notes
+    -----
+
+    The constructor can raise a RuntimeError if the constraint is deemed infeasible.
```

Comment:
Is this true? For SAGE there's a specific way that a constraint can be deemed infeasible a-priori.